### DISCUSSION TITLE
How to use KRaft?

### DISCUSSION THREAD

**Discussion by BryanDollery** [strimzi/strimzi-kafka-operator#7672] [Q&A] [answered]
I'm wondering how to create topics when using the KRaft mode?

I've watched the strimzi vlog on the subject, which doesn't help very much. It uses auto-creation of topics to get its job done (and it shows an error message despite the fact that it works).

I really want to use the crds to create my topics, because I want to control the partitions, replicas, and compacted status. How do I do that without the topic operator? Or, do I have to do it programmatically for now?

**Comment by scholzj** [comment, marked as answer]
You can do it programatically using the Kafka Admin API or the Kafka CLI tools which are based on it (such as `kafka-topic.sh`). The Topic Operator is unfortunately not supported in KRaft mode at this point.

**Reply by BryanDollery** [reply]
Thanks for taking the time to reply. Is there a roadmap for putting this in production? I guess it depends on when kraft makes it into production in kafka, but I imagine that there will be a lot of demand for this capability.

**Reply by scholzj** [reply]
It remains to be seen. This is quite complicated without ZooKeeper, because the KRaft metadata are not as easily accessible to the users and they are quite important for managing the topics.

**Comment by ignaciogonzalezsomoza** [comment]
Hello. Starting from a scenario in which we are already using a previous version of strimzi (0.27.0), what would be the strimzi roadmap or plan to upgrade to the kraft version starting from an existing cluster?

**Reply by scholzj** [reply]
Sorry, I'm not sure I understand what you are asking about.

**Comment by ignaciogonzalezsomoza** [comment]
I mean if there is going to be a plan to upgrade previous versions of strimzi in a production environment to new versions that include kraft and how this upgrade should be done.

**Reply by scholzj** [reply]
Well, this is not supported today in Kafka, so not sure if anyone can give you a really clear answer. The general expectation is that there will be some bridge version (possibly more then one, but certainly not many) to which you will have to upgrade and do the migration to KRaft. And then possibly continue upgrading to newer versions. This AFAU applies to Kafka and will likely apply to Strimzi as well.
